src/views/seeds/router.ts

```diff
@@ -0,0 +1,70 @@
+import type { BaseUrl, Project } from "@httpd-client";
+import type { SeedsParams, LoadedRoute } from "@app/lib/router/definitions";
+import type { WeeklyActivity } from "@app/lib/commit";
+
+export interface ProjectActivity {
+  project: Project;
+  activity: WeeklyActivity[];
+}
+
+import { HttpdClient } from "@httpd-client";
+import { extractBaseUrl } from "@app/lib/utils";
+import { loadProjectActivity } from "@app/lib/commit";
+
+const PROJECTS_PER_PAGE = 10;
+
+export async function loadProjects(
+  page: number,
+  baseUrl: BaseUrl,
+): Promise<{
+  count: number;
+  projects: ProjectActivity[];
+}> {
+  const api = new HttpdClient(baseUrl);
+
+  const [nodeStats, projects] = await Promise.all([
+    api.getStats(),
+    api.project.getAll({ page, perPage: PROJECTS_PER_PAGE }),
+  ]);
+
+  const results = await Promise.all(
+    projects.map(async project => {
+      const activity = await loadProjectActivity(project.id, baseUrl);
+      return {
+        project,
+        activity,
+      };
+    }),
+  );
+
+  return {
+    count: nodeStats.projects.count,
+    projects: results,
+  };
+}
+
+export async function loadSeedRoute(params: SeedsParams): Promise<LoadedRoute> {
+  const baseUrl = extractBaseUrl(params.hostnamePort);
+  const api = new HttpdClient(baseUrl);
+  try {
+    const nodeInfo = await api.getNodeInfo();
+    const projectPageIndex = 0;
+    const results = await loadProjects(projectPageIndex, baseUrl);
+    return {
+      resource: "seeds",
+      params: {
+        projectPageIndex: projectPageIndex + 1,
+        baseUrl,
+        nid: nodeInfo.node.id,
+        version: nodeInfo.version,
+        projects: results.projects,
+        projectCount: results.count,
+      },
+    };
+  } catch (error: any) {
+    return {
+      resource: "seedsLoadError",
+      params: { hostnamePort: params.hostnamePort, stackTrace: error.stack },
+    };
```

Comment:
Here are some thoughts around error handling. Nothing that we need to change now but we should keep an eye on this and explore different solutions.
* Should we have only one error route and have the route loaders generate the inputs
* We need to show `error.message` as well
* We should probably unfold [error cause chains](https://github.com/tc39/proposal-error-cause#chaining-errors) in the error view to provide more detailed information
* We should make use of error cause chains to provide more detailed information